src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrPusher.cs

```diff
@@ -0,0 +1,216 @@
+
+        var commits = await GetRepositoriesCommits(commitsSearchArguments, gitHubApiPat, cancellationToken);
+        
+        if (commits == null || commits.Data == null)
+        {
+            _logger.LogError("The Graphql query is not in the correct format.");
+            return false;
+        }
+
+        foreach (var pair in commits.Data)
+        {
+            if (pair.Value == null)
+            {
+                var commitArgs = commitsSearchArguments.First(r => SanitizeName(r.RepoName) == pair.Key);
+                _logger.LogError($"Repository {commitArgs.RepoName} with owner {commitArgs.RepoOwner} couldn't be found on GitHub.");
+                return false;
+            }
+
+            if (pair.Value.Object == null)
+            {
+                var commitArgs = commitsSearchArguments.First(r => SanitizeName(r.RepoName) == pair.Key);
+                _logger.LogError($"The commit {commitArgs.Sha} couldn't be found in public repo {commitArgs.RepoName} with owner {commitArgs.RepoOwner} on GitHub.");
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private async Task<CommitsQueryResult?> GetRepositoriesCommits(
+        IEnumerable<CommitSearchArguments> commits,
+        string gitHubApiPat,
+        CancellationToken cancellationToken)
+    {
+        var queries = commits
+            .Select(c => $@"{SanitizeName(c.RepoName)}: repository(name: \""{c.RepoName}\"", owner: \""{c.RepoOwner}\""){{object(oid: \""{c.Sha}\""){{ ... on Commit {{id}}}}}}");
+        var query = string.Join(", ", queries);
+        var body = @"{""query"" : ""{" + query + @"}""}";
+
+        HttpResponseMessage? response = null;
+        var url = "https://api.github.com/graphql";
+
+        _httpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue("Darc", null));
+
+        using HttpRequestMessage message = new(HttpMethod.Post, url);
+        message.Content = new StringContent(body, Encoding.UTF8, "application/json");
+        message.Headers.Authorization = new AuthenticationHeaderValue("Token", gitHubApiPat);
+
+        try
+        {
+            response = await _httpClient.SendAsync(message, cancellationToken);
+            response.EnsureSuccessStatusCode();
+        }
+        catch (HttpRequestException)
+        {
+            _logger.LogError("Cannot connect to GraphQl api");
+            throw;
+        }
+
+        var content = response.Content.ReadAsStringAsync(cancellationToken).Result;
+        var settings = new JsonSerializerOptions
+        {
+            PropertyNameCaseInsensitive = true,
+        };
+
+        var result = JsonSerializer.Deserialize<CommitsQueryResult>(content, settings);
+        return result;
+    }
+
+    private CommitSearchArguments GetCommitSearchArguments(string uri, string sha)
+    {
+        var gitHubUrlPrefix = "https://github.com";
+        var azDoUrlPrefix = "https://dev.azure.com";
```

Comment:
Can we have this somewhere in `Constants` maybe? Maybe it exists somewhere already? If not, it probably should.

Reply:
You could possible also utilize `GitRepoTypeParser` for this